### PR TITLE
Cypress testing & refactor

### PR DESCRIPTION
**Pull request by dkwon1223**
# Pull Request 

## Description of Changes

This update addresses cypress testing the user flow of viewing drink details and arriving at a 404 error view when navigating to a non-existent URL route. It also revises the drink grid cypress testing and repairs the fixture functionality.
In addition, it implements some refactoring including a new API call that performs a PATCH request on the isFavorite property of the drinks in their respective data on the backend when it is toggled through the UI.

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🔮 New feature (non-breaking change which adds functionality)
- [ ] 📖 Documentation Update
- [ ] Other (please describe):

## Checklist ✅

- [x] I have performed a self-review of my own code.
- [ ] I have made notes of my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove the new feature works if necessary.
- [x] All tests are passing.

### CODE REVIEW

File: cypress/fixtures/cocktails.json [hunk 1]
```diff
@@ -1,82 +1,80 @@
...
+            "2 oz Blackberry Vodka",
+            "1/2 oz Cointreau",
+            "1 oz Lemon Juice",
+            "Blackberries for garnish",
+            "Ice"
+        ],
+        "directions": {
+            "step1": "In a shaker, muddle a handful of blackberries.",
+            "step2": "Add blackberry vodka, Cointreau, lemon juice, and ice. Shake well.",
+            "step3": "Strain into a chilled martini glass and garnish with fresh blackberries."
```
Comment (crotteau) [inline cypress/fixtures/cocktails.json:15]: Why are the spaces below added?

File: src/apiCalls.js [hunk 2]
```diff
@@ -7,6 +7,22 @@ async function getRecipes() {
...
+            'Content-Type':'application/json'
+        },
+        body: JSON.stringify({
+            "isFavorite": newBoolean
+        })
+    })
+    .then(response => {
+        console.log(response.json());
+    })
+}
```
Comment (crotteau) [inline src/apiCalls.js:23]: nice work adding in patch!

**Review by crotteau — APPROVED** [review]
Nice work adding in a PATCH and error handling

**Review by lalonggone — APPROVED** [review]
Nice intercepts and patch!!!

**Pull request merged**